src/Functions/test/Functions.Scenario.Tests.ps1

```diff
@@ -922,4 +1014,396 @@ Describe 'Functions End to End Tests' {
                 Remove-AzFunctionAppPlan -Force -ErrorAction SilentlyContinue
         }
     }
+
+    It "Create a function app with custom app settings and 'SystemAssigned' managed identity " {
+
+        $functionName = "Func99-Windows-PowerShell-6-2-546452530"
+        $appSetting = @{}
+        $appSetting.Add("MyAppSetting1", 98765)
+        $appSetting.Add("MyAppSetting2", "FooBar")
+
+        try
+        {
+            New-AzFunctionApp -Name $functionName `
+                              -ResourceGroupName $resourceGroupNameWindowsPremium `
+                              -PlanName $planNameWorkerTypeWindows `
+                              -StorageAccount $storageAccountWindows  `
+                              -Runtime PowerShell `
+                              -IdentityType SystemAssigned `
+                              -AppSetting $appSetting
+
+            $functionApp = Get-AzFunctionApp -Name $functionName -ResourceGroupName $resourceGroupNameWindowsPremium
+            $functionApp.OSType | Should -Be "Windows"
+            $functionApp.Runtime | Should -Be "PowerShell"
+            $functionApp.IdentityType | Should -Be "SystemAssigned"
+
+            foreach ($appSettingName in $appSetting.Keys)
+            {
+                $functionApp.ApplicationSettings[$appSettingName] | Should Be $appSetting[$appSettingName]
+            }
+        }
+        finally
+        {
+            Get-AzFunctionApp -Name $functionName -ResourceGroupName $resourceGroupNameWindowsPremium -ErrorAction SilentlyContinue |
+                Remove-AzFunctionApp -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    It "Create a function app with 'UserAssigned' managed identity " {
+
+        # Make sure user identiy is available
+        $userIdentityId | Should Not Be $null
+
+        $functionName = "Func99-Windows-PowerShell-6-2-1009551322"
```

Comment:
sma eas above - any names of entities created int the test or rrewuired by the test should be set in the setup script, and written to $env hashtable

Reply:
Fixed.